src/jdk.naming.dns/share/classes/com/sun/jndi/dns/DnsClient.java

```diff
@@ -471,7 +474,9 @@ private byte[] doUdpQuery(Packet pkt, InetAddress server,
                             return cachedMsg;
                         }
                     }
-                    timeoutLeft = pktTimeout - ((int) (end - start));
+                    long elapsedMillis = TimeUnit.NANOSECONDS
+                                                 .toMillis(System.nanoTime() - start);
+                    timeoutLeft = pktTimeout - Math.clamp(elapsedMillis, 0, Integer.MAX_VALUE);
```

Comment:
```suggestion
                    timeoutLeft = pktTimeout - elapsedMillis;
```
Now that timeoutLeft is a long we have no reason to clamp at Integer.MAX_VALUE.

Reply:
Thanks fixed in [3abb782](https://github.com/openjdk/jdk/pull/20892/commits/3abb782904cf80d2ed5da70266cdfeea05b1bd2f)